plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -34,17 +47,60 @@
   public static final String OAUTH_CLIENT_ID = "@oauth.client.id@";
   public static final String OAUTH_CLIENT_SECRET = "@oauth.client.secret@";
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email",
+      "https://www.googleapis.com/auth/cloud-platform"
   ));
 
   /**
    * Returns the credential of the active user. If there is no active user, returns {@code null}.
    */
   // Should probably be synchronized properly.
   // TODO(chanseok): consider returning a String JSON (i.e., hide Credential)
-  public Credential getActiveCredential() {
-    return null;
+  public Credential getActiveCredential() throws IOException {
+    GoogleTokenResponse authResponse = logIn();
+    return createCredential(authResponse);
+  }
+
+  private GoogleTokenResponse logIn() throws IOException {
+    // 1. Open a browser that takes care of the login.
+    GoogleAuthorizationCodeRequestUrl requestUrl = new GoogleAuthorizationCodeRequestUrl(
+        OAUTH_CLIENT_ID, GoogleOAuthConstants.OOB_REDIRECT_URI, OAUTH_SCOPES);
+
+    try {
+      IWorkbenchBrowserSupport browserSupport = PlatformUI.getWorkbench().getBrowserSupport();
+      browserSupport.getExternalBrowser().openURL(requestUrl.toURL());
+    } catch (PartInitException pie) {
+      // TODO(chanseok): display error message to user
+      return null;
+    }
+
+    // 2. Show a dialog to get a verification code returned from successful browser login.
+    InputDialog dialog = new InputDialog(Display.getCurrent().getActiveShell(),
```

Comment:
A good approach is to take an `org.eclipse.jface.window.IShellProvider` rather than requiring an explicit Shell.  There's a wrapper IShellProvider class `org.eclipse.jface.window.SameShellProvider` that returns its provided shell.